From ed916ff21629f3e91ee987552d778b1a65d66702 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Sat, 17 Jun 2023 13:01:43 +0100
Subject: [PATCH] effects: Make OpenGL context current before deleting
 framebuffer

When we delete a window we unredirect, when we unredirect, we delete the
framebuffer which requires an openGL context.

handleWindowDeleted is the entry point from workspace events to effects
code, so it's made current here.

BUG: 444665
BUG: 471139


(cherry picked from commit c5a29b4b66c001c49c5bcf9aa9250d8322eefbbd)
---
 src/libkwineffects/kwinoffscreeneffect.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libkwineffects/kwinoffscreeneffect.cpp b/src/libkwineffects/kwinoffscreeneffect.cpp
index 82abea675c1..d1bd2921576 100644
--- a/src/libkwineffects/kwinoffscreeneffect.cpp
+++ b/src/libkwineffects/kwinoffscreeneffect.cpp
@@ -238,6 +238,7 @@ void OffscreenEffect::handleWindowDamaged(EffectWindow *window)
 
 void OffscreenEffect::handleWindowDeleted(EffectWindow *window)
 {
+    effects->makeOpenGLContextCurrent();
     unredirect(window);
 }
 
-- 
GitLab

